### PR TITLE
marking services as deprecated

### PR DESCRIPTION
**Pull request by stefanmcshane**
- Marking service calls as deprecated. Some of these will require work before deleting

### CODE REVIEW

**Review by d-g-town — APPROVED** [review]

File: porter/v1/cluster.proto [hunk 1]
```diff
@@ -38,3 +38,65 @@ enum EnumCloudProvider {
   ENUM_CLOUD_PROVIDER_GCP = 2;
   ENUM_CLOUD_PROVIDER_AZURE = 3;
 }
+
+// // AWSCredentials are all the hops required to access a target account
```
Comment (stefanmcshane) [inline porter/v1/cluster.proto:42]: This will be removed in an immediate followup PR

File: porter/v1/cluster_control_plane.proto [hunk 2]
```diff
@@ -221,22 +166,114 @@ service ClusterControlPlaneService {
...
   rpc ListImagesForRepository(ListImagesForRepositoryRequest) returns (ListImagesForRepositoryResponse) {}
+
+  // DockerConfigFileForRegistry returns a stringified config.json for accessing a given registry.
+  // Deprecated. Use TokenForRegistry instead.
+  rpc DockerConfigFileForRegistry(DockerConfigFileForRegistryRequest) returns (DockerConfigFileForRegistryResponse) {
+    option deprecated = true;
+  }
+
+  // ECRTokenForRegistry returns a docker-compatible token for accessing a given ECR registry. Deprecated. Use TokenForRegistry instead.
+  rpc ECRTokenForRegistry(ECRTokenForRegistryRequest) returns (ECRTokenForRegistryResponse) {
```
Comment (d-g-town) [inline porter/v1/cluster_control_plane.proto:185]: This cannot be deleted before TokenForRegistry supports AWS and we edit `RegistryGetECRTokenHandler` to call that endpoint

Comment (stefanmcshane) [reply porter/v1/cluster_control_plane.proto:185]: Will add to the comment. Plan here is to deprecated everything now, then update the endpoints after GKE is done.

File: porter/v1/cluster_control_plane.proto [hunk 3]
```diff
@@ -221,22 +166,114 @@ service ClusterControlPlaneService {
...
+    option deprecated = true;
+  }
+
+  // AssumeRoleCredentials should be used vary sparingly, and ONLY for replacing AWS Integrations which have no workaround on the Porter API.
+  // This endpoint returns temporary AWS credentials for a given AWS Account ID, and should not be expanded further to allow specifc role selection without being tied to a project and cluster.
+  // Deprecated. This is no longer needed once ListRepositoriesForRegistry supports AWS.
```
Comment (d-g-town) [inline porter/v1/cluster_control_plane.proto:191]: We will also need to support ListImages and CreateRepository to fully delete this endpoint.

**Pull request merged**